twyla/kubedeploy/kubectl.py

```diff
@@ -0,0 +1,73 @@
+import json
+import functools
+import subprocess
+
+
+class KubectlCallFailed(Exception):
+    pass
+
+
+class Kubectl:
+    def __init__(self):
+        self.exe = 'kubectl'
+        self.args = [
```

Comment:
I don't understand the reason behind the `args` attribute and modifying it with every command. `_make_command` uses it to build the command, and each different method for calling kubectl passes it in by setting the `args` attribute. Wouldn't it make sense to just pass it to `_make_command` as an argument, and have different attribute like `default_args` that is used when `_make_command` does not receive any arguments?

Reply:
My original plan was to have a set of fixed arguments like `'-o', 'json'` but that approach has basically failed because not all commands provide json output. Will redo that part to make more sense.